adapters/repos/db/index.go

```diff
@@ -782,8 +782,14 @@ func (i *Index) objectSearch(ctx context.Context, limit int, filters *filters.Lo
 		}
 	}
 
-	outObjects := make([]*storobj.Object, 0, len(shardNames)*limit)
-	outScores := make([]float32, 0, len(shardNames)*limit)
+	// use initially limited cap for reference (effectively unlimited) results
+	cap := int(config.DefaultQueryMaximumResults)
+	if !additional.ReferenceQuery {
+		cap = len(shardNames) * limit
```

Comment:
I think changing the limit value to `math.MaxInt` might cause a serious issue` if !additional.ReferenceQuery`:
`math.MaxInt` is 2^63-1 on 64-bit architecture which is a huge number. 
Therefore, `len(shardNames) * limit` might result in an OOM exception or integer overflow (see line 819).